docs/proposals/20191017-control-plane.md

```diff
@@ -0,0 +1,495 @@
+    - [Upgrade Strategy](#upgrade-strategy)
+  - [Alternatives](#alternatives)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+The lexicon used in this document is described in more detail [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).  Any discrepancies should be rectified in the main Cluster API glossary.
+Stacked Control Plane / Stacked etcd: Control plane node where etcd is colocated with the Kubernetes API server, and is running as a static pod.
+
+### References
+
+[Kubernetes Control Plane Management Overview](https://docs.google.com/document/d/1nlWKEr9OP3IeZO5W2cMd3A55ZXLOXcnc6dFu0K9_bms/edit#)
+
+## Summary
+
+Management of the control plane is currently handled in Cluster API at the infrastructure provider level (although many providers have converged on using the same implementation), based on the special annotation of standard Cluster API machines which then changes the machine bootstrapping logic within a particular infrastructure provider.
+The purpose of this document is to instead treat machine-based control plane management as a first-class component that is part of the API and Cluster, where multiple Cluster API machines will belong to a single control plane object, with its own infrastructure independent provider. The control plane covered by this document is defined as consisting of the Kubernetes API server, scheduler, controller manager, DNS and proxy services, and the underlying etcd datastore.
+
+## Motivation
+
+### Goals
+
+- To establish new resource types for control plane management
+- To support single node and multiple control plane instances
+- To enable declarative orchestrated control plane upgrades
+- To provide a default machine-based implementation using kubeadm
+
+#### Additional goals of the default kubeadm machine-based Implementation
+
+- To provide a kubeadm-based implementation that is infrastructure provider agnostic
+- To manage kubeadm-based "stacked etcd" control plane deployments
+- To enable scaling of the number of nodes in a control plane deployment
+- To support pre-existing, user-managed external etcd clusters
+- To support only user-initiated remediation:
+  E.g. user deletes a Machine. Control Plane Provider reconciles by removing the corresponding etcd member and updating related metadata
+
+### Non-Goals / Future Work
+
+Non-Goals listed in this document are intended to scope bound the current v1alpha3 implementation and are subject to change based on user feedback over time.  
+
+- To manage non-machine based topologies, e.g.
+  - Pod based control planes; these can be managed via standard kubernetes objects.
+  - Non-node control planes (i.e. EKS, GKE, AKS); these can be managed via the respective APIs.
+- To manage CA certificates outside of what is provided by Kubeadm bootstrapping
+- To manage etcd clusters in any topology other than “stacked etcd.”
```

Comment:
nit: replace "smart quotes" with normal quotes. (or just remove quotes entirely, since the term is in the glossary)